streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -480,6 +482,11 @@ public Set<String> makeReady(final Map<String, InternalTopicConfig> topics) {
                     .filter(e -> topicsNotReady.contains(e.getKey()))
                     .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
             final Set<NewTopic> topicsToCreate = computeTopicsToCreate(notReadyTopicsMap, tempUnknownTopics);
+
+            topicsNotReady.retainAll(tempUnknownTopics);
+            topicsNotReady.retainAll(notReadyTopicsMap.keySet());
+            topicsToCreate.forEach(topic -> topicsNotReady.add(topic.name()));
```

Comment:
This seems to be unnecessarily complex? We first remove topics, and in the third line add back topics... So it seem we are removing topic we should not have remove, just to add them back?

In the end, we should only remove the topics from `topicNotReady` which do already exist.

Did you see my comment from my previous review: https://github.com/apache/kafka/pull/20326#discussion_r2512435233

I think, if we modify `topicsNotReady` there (ie, `topicMap` which is the corresponding parameter name inside the `computeTopicsToCreate` for which we pass in `topicsNotReady`), it should fix it, allowing us to avoid the code complexity here?